### PR TITLE
Allow a comma-list of values per cond case

### LINKED ISSUE #160: Allow a comma-list of values per cond case
E.g.:
```
x cond (
    1: "lo",
    (2, 3): "med",
    *: "hi"
)
```

The parentheses are required in order to avoid confusion in single-line usage. Consider `x cond (1: -1, 2, 3: 0, *: 1)` vs `x cond (1: -1, (2, 3): 0, *: 1)`.

Only allow this for `x cond (...)` syntax. For `cond (...)` syntax, multiple cases can simply be `||`-ed together.

### PR DESCRIPTION
**Pull request by ericzhang6222**
Fixes #160.

Changes proposed in this pull request to support following syntax:
- `cond ((1 > 0 && 3 > 2): 1, (2 > 3) || (1 < 0): 2, *:1 + 2)`
- `let a = cond (cond (1 > 2 : 1, * : 11) < 2 : 1, 2 < 3: 2, *:1 + 2);a * 10`
- `let a = 2; a cond (1 :1 + 10, (2,3) : 2, *:1 + 2)`
- `let a = 2;
	a cond (
		1:"lo",
		(2,3): "med",
		*: "hi")`
- `let a = 2; a cond ( a cond ((1,2) : 1): "A", (2, 3): "B", *: "C")`
- `let a = 1; a cond ( cond (2 > 1 : 1): "A", (2, 3): "B", *: "C")`
- `let a = 1; cond ( a cond (1 : 1) : "A", 2: "B", *: "C")`

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: syntax/arrai.wbnf [hunk 1]
```diff
@@ -30,7 +30,7 @@ expr   -> C* amp="&"* @ C* arrow=(
         | C* "{" C* set=(elt=@:",",?) "}" C*
         | C* "{" C* dict=((key=@ ":" value=@):",",?) "}" C*
         | C* cond=("cond" "(" (key=@ ":" value=@):",",? ("*" ":" f=expr ","?)? ")") C*
-        | C* cond=(("(" control_var=expr ")" | IDENT)? C* "cond" "(" (key=@ ":" value=@):",",? ("*" ":" f=expr ","?)? ")") C*
+        | C* cond=(("(" control_var=expr ")" | IDENT)? C* "cond" "(" (key=(("(" @:"," ")") | @) ":" value=@):",",? ("*" ":" f=expr ","?)? ")") C*
```
Comment (marcelocantos) [inline syntax/arrai.wbnf:33]:
```suggestion
        | C* cond=(("(" control_var=expr ")" | IDENT)? C* "cond" "(" (key=(("(" @:",", ")") | @) ":" value=@):",",? ("*" ":" f=expr ","?)? ")") C*
```

File: syntax/compile.go [hunk 2]
```diff
@@ -495,7 +495,13 @@ func (pc ParseContext) compileExpr(c ast.Children) rel.Expr {
 		if len(c) == 1 {
 			return pc.CompileExpr(c[0].(ast.Branch))
 		}
-		panic("too many expr children")
```
Comment (ericzhang6222) [inline syntax/compile.go:498]: I am not sure if this change will break the initial design, can you help to double confirm?

File: syntax/parser.go [hunk 3]
```diff
@@ -44,7 +44,7 @@ expr   -> C* amp="&"* @ C* arrow=(
         | C* "{" C* set=(elt=@:",",?) "}" C*
         | C* "{" C* dict=((key=@ ":" value=@):",",?) "}" C*
         | C* cond=("cond" "(" (key=@ ":" value=@):",",? ("*" ":" f=expr ","?)? ")") C*
-        | C* cond=(("(" control_var=expr ")" | IDENT)? C* "cond" "(" (key=@ ":" value=@):",",? ("*" ":" f=expr ","?)? ")") C*
+        | C* cond=(("(" control_var=expr ")" | IDENT)? C* "cond" "(" (key=(("(" @:"," ")") | @) ":" value=@):",",? ("*" ":" f=expr ","?)? ")") C*
```
Comment (marcelocantos) [inline syntax/parser.go:47]: Tag this file as generated via git attributes. https://github.com/github/linguist#generated-code

File: syntax/arrai.wbnf (hunk 1 above)
Comment (marcelocantos) [inline syntax/arrai.wbnf:33]: This should be much lower precedence. Put it right above or below the original `if...else` construct. Just the `x cond (...)` syntax, mind you. The `cond (...)` form can stay where it is.

**Review by marcelocantos — CHANGES_REQUESTED** [review]

File: syntax/compile.go (hunk 2 above)
Comment (marcelocantos) [reply syntax/compile.go:498]: You may be able to avoid this hack by moving the logic into `compileCond`.

File: syntax/compile.go [hunk 4]
```diff
@@ -495,7 +495,13 @@ func (pc ParseContext) compileExpr(c ast.Children) rel.Expr {
...
+
+		var elements []rel.Expr
+		for _, e := range c {
+			expr := pc.CompileExpr(e.(ast.Branch))
+			elements = append(elements, expr)
+		}
+		return rel.NewArrayExpr(c.Scanner(), elements...)
```
Comment (marcelocantos) [inline syntax/compile.go:504]: You may be able to move this logic into `compileCond`.

**Review by marcelocantos — APPROVED** [review]

**Pull request merged**